OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -133,21 +139,87 @@ static int HashList<T>(List<T> xs)
 			return hash;
 		}
 
-		List<CPos> EvalPath()
+		// NOTE TO REMOVE: This previously used mobile.ToCell. actor.Location
+		// and mobile.ToCell are exactly the same always, and using
```

Comment:
Done. Created a new property inside mobile called "CurrentLocation". It's a replica of ToCell but without that mentioned overhead. Semantics are important to me :)